store/src/java/com/zimbra/cs/service/mail/GetContactFrequency.java

```diff
@@ -35,21 +37,27 @@ public Element handle(Element request, Map<String, Object> context)
         Account acct = getRequestedAccount(zsc);
         GetContactFrequencyRequest req = zsc.elementToJaxb(request);
         String contactEmail = req.getEmail();
-        String freqByCSV = req.getFrequencyBy();
+        String freqByOptions = req.getFrequencyBy();
```

Comment:
What happens if this (or offsetInMinutes) is null?

Reply:
If the offset is null we calculated the offset from UTC to the preferred timezone set by the user. This is handled on line 56 and in ContactAnalytics class. The 'FrequencyBy' is a required option, so I think an exception would be raised.